astropy/io/misc/hdf5.py

```diff
@@ -259,7 +260,9 @@ def write_table_hdf5(table, output, path=None, compression=False,
         raise Exception("h5py is required to read and write HDF5 files")
 
     if path is None:
-        raise ValueError("table path should be set via the path= argument")
+        path = '__astropy__'
```

Comment:
Maybe leave an in-line comment explaining that this was an arbitrary choice, so we know the reasoning of hardcoding this particular value, for future reference.

Reply:
Done